diff --git a/include/fastdds/rtps/transport/TransportDescriptorInterface.h b/include/fastdds/rtps/transport/TransportDescriptorInterface.h
index 1497077..2b880cc 100644
--- a/include/fastdds/rtps/transport/TransportDescriptorInterface.h
+++ b/include/fastdds/rtps/transport/TransportDescriptorInterface.h
@@ -17,9 +17,7 @@
 
 #include <fastrtps/fastrtps_dll.h>
 
-#ifdef _WIN32
 #include <cstdint>
-#endif // ifdef _WIN32
 #include <vector>
 #include <string>
 
diff --git a/include/fastrtps/types/TypeNamesGenerator.h b/include/fastrtps/types/TypeNamesGenerator.h
index a8f45e0..d492bfe 100644
--- a/include/fastrtps/types/TypeNamesGenerator.h
+++ b/include/fastrtps/types/TypeNamesGenerator.h
@@ -5,6 +5,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 namespace eprosima {
 namespace fastrtps {
